test/buildConnectorsJson/generateSchemaFromJs.js

```diff
@@ -232,6 +233,144 @@ describe('#generateSchemaFromJs', function () {
 		assert.equal(output.properties.deepData.items.properties.subObject.properties.color.default, 'red');
 	});
 
-});
+	it('should recursively generate schema for deep arrays', function () {
+		var output = generateSchemaFromJs({
+			deepData: {
+				type: 'array',
+				items: {
+					type: 'array',
+					items: {
+						type: 'array',
+						items: {
+							type: 'string'
+						}
+					}
+				}
+			},
+			deepDataX: {
+				type: 'array',
+				items: {
+					type: 'object',
+					additionalProperties: true
+				}
+			}
+		});
+
+		assert.equal(output.properties.deepData.type, 'array');
+		assert.equal(output.properties.deepData.items.type, 'array');
+		assert.equal(output.properties.deepData.items.title, 'Item');
+		assert.equal(output.properties.deepData.items.items.type, 'array');
+		assert.equal(output.properties.deepData.items.items.items.type, 'string');
+		assert.equal(output.properties.deepDataX.items.type, 'object');
+
+	});
+
+	it('should add expand "oneOf" property properly', function () {
+		var outputObject = generateSchemaFromJs({
+			body_type: {
+				title: 'Body Type',
+				type: 'object',
+				additionalProperties: false,
+				oneOf: [
+
+					{
+						title: 'raw',
+						type: 'object',
+						additionalProperties: false,
+						properties: {
+							id: {
+								type: 'string',
+								default: 'raw',
+								format: 'hidden'
+							},
+							content: {
+								type: 'string',
+								title: 'raw',
+								format: 'code'
+							}
+						},
+						random: 'random'
+					},
+
+					{
+						title: 'form-urlencoded',
+						type: 'object',
+						additionalProperties: false,
+						properties: {
+							id: {
+								type: 'string',
+								default: 'form-urlencoded',
+								format: 'hidden'
+							},
+							content: {
+								type: 'object',
+								title: 'form-urlencoded',
+								additionalProperties: true
+							}
+						}
+					},
+
+					{
+						random: 'random'
+					},
+
+					{
+						type: 'string'
+					}
+				]
+			}
+		});
+
+		assert.equal(_.isArray(outputObject['properties']['body_type']['oneOf']), true);
```

Comment:
Don't worry about changing these - but you should use dot notation rather than arrays where possible